aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -233,23 +233,21 @@ static void norm_kernel_tensor_iterator_impl(
   // the type `acc_t`. When `scalar_t` is complex, `acc_t` is the downgraded
   // real number type. Otherwise, `acc_t` and `scalar_t` are the same type.
   if (val == 0) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
-      using acc_t = typename scalar_value_type<scalar_t>::type;
-      binary_kernel_reduce(
-        iter,
-        NormZeroOps<scalar_t, acc_t>(),
-        acc_t(0)
-      );
-    });
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+        kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
+          using real_type = typename scalar_value_type<scalar_t>::type;
+          using real_acc_t = at::opmath_type<real_type>;
+          binary_kernel_reduce(
+              iter, NormZeroOps<scalar_t, real_acc_t>(), real_acc_t(0));
+        });
   } else if (val == 1) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
-      using acc_t = typename scalar_value_type<scalar_t>::type;
-      binary_kernel_reduce(
-        iter,
-        NormOneOps<scalar_t, acc_t>(),
-        acc_t(0)
-      );
-    });
+    AT_DISPATCH_FLOATING_TYPES_AND2(
+        kHalf, kBFloat16, iter.input_dtype(), "norm_cpu", [&] {
+          using real_type = typename scalar_value_type<scalar_t>::type;
+          using real_acc_t = at::opmath_type<real_type>;
+          binary_kernel_reduce(
+              iter, NormOneOps<scalar_t, real_acc_t>(), real_acc_t(0));
+        });
   } else if (val == 2) {
```

Comment:
Not necessarily in this PR, we may want to extend this to the vectorised case for `val == 2`.

Reply:
We do need to extend this to the vectorized case, after all, this is the most commonly called norm.